waku/node/waku_node.nim

```diff
@@ -194,12 +194,15 @@ proc connectToNodes*(
 
 ## Waku Metadata
 
-proc mountMetadata*(node: WakuNode, clusterId: uint32): Result[void, string] =
+proc mountMetadata*(
+    node: WakuNode, clusterId: uint32, shards: seq[uint16] = @[]
+): Result[void, string] =
   if not node.wakuMetadata.isNil():
     return err("Waku metadata already mounted, skipping")
 
   let metadata = WakuMetadata.new(clusterId, node.enr, node.topicSubscriptionQueue)
-
+  for shard in shards:
+    metadata.shards.incl(uint32(shard))
```

Comment:
I think this change is not necessary and might be problematic.
The shards are added to metadata in `WakuMetadata.new`by using the parameter `node.enr`

If the node used in the Canary is subscribed to the right shards, then they should be in the ENR and added from there

Reply:
Thanks Gabriel !

Then i should try to add this shards information in ENR.